### PR TITLE
Code review cleanup items and moved some items into LEGACY_BACKEND

### PR DESCRIPTION
**Pull request by briansull**
From Codereview feedback:
Use BAD_VAR_NUM in a couple places instead of (unsigned)-1
In struct InitVarDscInfo
Renamed hasRetBuf to hasRetBufArg

In struct RegState:
Moved field rsCurRegArmNum into LEGACY_BACKEND ifdef
Removed instance field RegState::rsMaxRegArgNum as it is unnecessary
Reordered the fields from largest to smallest

In struct GenTreeCall node:
Moved field gtCallRegUsedMask into LEGACY_BACKEND ifdef

Changes to genRegArgNext to work correct for X64/UNIX (where it is not currently used)
Document the behavaior when given REG_ARG_LAST

### CODE REVIEW

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Comment by briansull** [comment]
@dotnet/jit-contrib PTAL

**Comment by mmitche** [comment]
@briansull clang compile error.  I added a new indication which should be able to highlight that without picking up other stuff (clang's error messages aren't very identifiable)

**Comment by mmitche** [comment]
@briansull  /mnt/resource/j/workspace/dotnet_coreclr/master/checked_freebsd_prtest/src/jit/regset.cpp:3495:8: error: extra tokens at end of #endif directive [-Werror,-Wextra-tokens]
#endif _TARGET_AMD64_

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT arm64 Checked

File: src/jit/regset.cpp [hunk 1]
```diff
@@ -3458,19 +3458,52 @@ bool                genIsProperRegPair(regPairNo regPair)
  *
  *  Given a register that is an argument register 
  *   returns the next argument register 
+ *
+ *  Note: that this method will return a non arg register 
+ *   when given REG_ARG_LAST
+ *
```
Comment (CarolEidt) [inline src/jit/regset.cpp]: I'm curious why you chose this behavior instead of returning REG_NA or something.

**Comment by CarolEidt** [comment]
I have a question - but the changes look good.

File: src/jit/regset.cpp (hunk 1 above)
Comment (briansull) [inline src/jit/regset.cpp]: It more that I am documenting the behavior not choosing it.

I tried adding an assert that  we were returning a valid argument register, but that failed.
I probably could also change it to return REG_NA and that would probably work.

**Comment by briansull** [comment]
@dotnet-bot   CentOS7.1 x64 Debug Build and Test

**Comment by mmitche** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build and Test (you missed the test bit)

**Pull request merged**